packages/backend/server/src/plugins/copilot/providers/openai.ts

```
@@ -461,29 +462,37 @@ export class OpenAIProvider extends CopilotProvider<OpenAIConfig> {
         });
 
-        const top = (logprobs?.[0]?.topLogprobs ?? []).reduce(
-          (acc, item) => {
-            acc[item.token] = item.logprob;
-            return acc;
-          },
-          {} as Record<string, number>
+        const topMap: Record<string, number> = (
+          logprobs?.[0]?.topLogprobs ?? []
+        ).reduce<Record<string, number>>(
+          (acc, { token, logprob }) => ({ ...acc, [token]: logprob }),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)